### PR TITLE
Batch credentials are too exposed #989

### PR DESCRIPTION
**Pull request by ryanemerson**

### CODE REVIEW

**Comment by Crumby** [comment]
@ryanemerson Is the secret with `cli.properties` created for each Batch? Could we create a single one altogether with the initial cluster that'd be used with all the batches targeting that particular cluster?

File: pkg/controller/infinispan/resources/secret/secret_controller.go [hunk 1]
```diff
@@ -75,16 +76,25 @@ func Add(mgr manager.Manager) error {
...
 	// If the operator secret does not already exist create it with generated identities
-	if !secretExists {
+	if secret == nil {
 		if err = s.createAdminIdentitiesSecret(); err != nil {
 			return reconcile.Result{}, err
 		}
+	} else {
+		// Patch secret to include any generated fields
+		updated, err := s.update(secret, func() error {
+			return s.addCliProperties(secret)
```
Comment (ryanemerson) [inline pkg/controller/infinispan/resources/secret/secret_controller.go:92]: @dmvolod I was envisaging that you could add an additional call to this update function and in `createSecret()` to a method that generates the monitoring `username` & `password` fields

Comment (dmvolod) [reply pkg/controller/infinispan/resources/secret/secret_controller.go:92]: Yes, cool, @ryanemerson 
We need to merge this PR before and I think this section will be permanent and never been deleted

**Review by Crumby — APPROVED** [review]

**Comment by dmvolod** [comment]
@ryanemerson you missed `patch` rights in Role for Secret

**Review by dmvolod — APPROVED** [review]

**Pull request merged**